californium-core/src/test/java/org/eclipse/californium/core/coap/RequestTest.java

```diff
@@ -67,8 +68,21 @@ public void testSetURISetsDestination() throws UnknownHostException {
 
 	@Test
 	public void testSetURISetsUriHostOption() throws UnknownHostException {
-		Request req = Request.newGet().setURI("coap://iot.eclipse.org");
-		assertThat(req.getOptions().getUriHost(), is("iot.eclipse.org"));
```

Comment:
That's a good questio, @jvermillard. The whole code around creating a `Request` from a URI could probably use an overhaul based on [Section 6.4 of the CoAP spec](https://tools.ietf.org/html/rfc7252#section-6.4) ...